src/store/uiState.js

```diff
@@ -0,0 +1,114 @@
+import mergeDeep from "@utils/merge-deep";
+
+const localStorageKey = "uiState";
+const version = 1;
+let initialize = false;
+
+const defaultState = {
+	pagination: {},
+	filter: {},
+	version,
+};
+
+const getDefaultState = () => {
+	if (localStorage.getItem(localStorageKey)) {
+		const uiState = JSON.parse(localStorage.getItem(localStorageKey));
+		if (uiState.version == version) {
+			Object.assign(defaultState, uiState);
+		} else {
+			uiState.version = version;
+		}
+	}
+	initialize = true;
+
+	return defaultState;
+};
```

Comment:
```suggestion
const getDefaultState = () => {
	const defaultState = {
		pagination: {},
		filter: {},
		version,
	};
	if (localStorage.getItem(localStorageKey)) {
		const uiState = JSON.parse(localStorage.getItem(localStorageKey));
		if (uiState.version == version) {
			Object.assign(defaultState, uiState);
		} else {
			uiState.version = version;
		}
	}
	initialize = true;

	return defaultState;
};
```
move the defaultState definition into the method to create and use a new object for defaultState each time. Otherwise there may be conflicts when resetting the store, because you had modified your defaultState.